flow-typed/react-reconciler.js

```diff
@@ -0,0 +1,303 @@
+// @flow
+
+type Fiber = any;
+type FiberRoot = any;
+
+type Deadline = {
+  timeRemaining: () => number
+};
+
+export type ReactNode =
+  | React$Element<any>
+  | ReactCall<any>
+  | ReactReturn<any>
+  | ReactPortal
+  | ReactText
+  | ReactFragment;
+
+type ReactFragment = ReactEmpty | Iterable<React$Node>;
+
+type ReactNodeList = ReactEmpty | React$Node;
+
+type ReactText = string | number;
+
+type ReactEmpty = null | void | boolean;
+
+type ReactCall<V> = {
+  $$typeof: Symbol | number,
+  type: Symbol | number,
+  key: null | string,
+  ref: null,
+  props: {
+    props: any,
+    // This should be a more specific CallHandler
+    handler: (props: any, returns: Array<V>) => ReactNodeList,
+    children?: ReactNodeList
+  }
+};
+
+type ReactReturn<V> = {
+  $$typeof: Symbol | number,
+  type: Symbol | number,
+  key: null,
+  ref: null,
+  props: {
+    value: V
+  }
+};
+
+type ReactPortal = {
+  $$typeof: Symbol | number,
+  key: null | string,
+  containerInfo: any,
+  children: ReactNodeList,
+  // TODO: figure out the API for cross-renderer implementation.
+  implementation: any
+};
+
+declare module 'react-reconciler' {
+  declare export type OpaqueHandle = Fiber;
+  declare export type OpaqueRoot = FiberRoot;
+  declare export type ExpirationTime = number;
+
+  declare export type HostConfig<T, P, I, TI, HI, PI, C, CC, CX, PL> = {
```

Comment:
"how many layers of generics you on?",
"like 4 or five my dude"
"You're like a little baby. watch this"

@gaearon single letter generic types seem fairly commmon in Flow. Is there a reason, that you know of, why slightly more descriptive convention isn't used, like `<TInstance>` or something a la C# ?

Reply:
I dunno, that's what I'd prefer too but @sebmarkbage didn't like it (I don't remember why).